packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -718,69 +520,58 @@ export const optionalChangeHandler: FieldChangeHandler<
+		change.childChange === undefined && change.valueReplace === undefined,
 
 	getNestedChanges,
 
-	createEmpty: () => ({ moves: [], childChanges: [] }),
-	getCrossFieldKeys: (_change) => [],
+	createEmpty: () => ({}),
+	getCrossFieldKeys,
 };
 
-function getNestedChanges(change: OptionalChangeset): NestedChangesIndices {
-	// True iff the content of the field changes in some way
-	const isFieldContentChanged =
-		change.valueReplace !== undefined && change.valueReplace.src !== "self";
-
-	// The node that is moved into the field (if any).
-	const nodeMovedIntoField = change.valueReplace?.src;
-
-	return change.childChanges.map(([register, nodeId]) => {
-		// The node is removed in the input context iif register is not self.
-		const inputIndex = register === "self" ? 0 : undefined;
-		const outputIndex =
-			register === "self"
-				? // If the node starts out as not-removed, it is removed in the output context iff the field content is changed
-					isFieldContentChanged
-					? undefined
-					: 0
-				: // If the node starts out as removed, then it remains removed in the output context iff it is not the node that is moved into the field
-					!areEqualRegisterIdsOpt(register, nodeMovedIntoField)
-					? undefined
-					: 0;
-		return [nodeId, inputIndex, outputIndex];
-	});
-}
+function getCrossFieldKeys(change: OptionalChangeset): CrossFieldKeyRange[] {
+	const keys: CrossFieldKeyRange[] = [];
+	if (change.valueReplace !== undefined) {
+		keys.push({
+			key: { ...change.valueReplace.dst, target: CrossFieldTarget.Source },
+			count: 1,
+		});
 
-function* relevantRemovedRoots(
-	change: OptionalChangeset,
-	relevantRemovedRootsFromChild: RelevantRemovedRootsFromChild,
-): Iterable<DeltaDetachedNodeId> {
-	const alreadyYielded = new RegisterMap<boolean>();
-
-	for (const [src] of change.moves) {
-		if (!alreadyYielded.has(src)) {
-			alreadyYielded.set(src, true);
-			yield nodeIdFromChangeAtom(src);
+		if (change.valueReplace.src !== undefined && change.valueReplace.src !== "self") {
+			keys.push({
+				key: { ...change.valueReplace.src, target: CrossFieldTarget.Destination },
+				count: 1,
+			});
 		}
 	}
 
-	for (const [id, childChange] of change.childChanges) {
-		// Child changes make the tree they apply to relevant unless that tree existed in the starting context of
-		// of this change.
-		if (id !== "self" && !alreadyYielded.has(id)) {
-			alreadyYielded.set(id, true);
-			yield nodeIdFromChangeAtom(id);
-		}
-		yield* relevantRemovedRootsFromChild(childChange);
+	return keys;
+}
+
+function getNestedChanges(change: OptionalChangeset): NestedChangesIndices {
+	if (change.childChange === undefined) {
+		return [];
 	}
 
-	const selfSrc = change.valueReplace?.src;
-	if (selfSrc !== undefined && selfSrc !== "self" && !alreadyYielded.has(selfSrc)) {
-		yield nodeIdFromChangeAtom(selfSrc);
+	return [[change.childChange, 0]];
+}
+
+/**
+ * Helper function for use in rebase which returns child change for the base changeset.
```

Comment:
The child change for the base changeset is just `baseChange.childChange` so this helper is providing something else. Is it trying to get the changeset for a specific node?